### PR TITLE
Fix the unhandled exception and test validation in manual tests

### PR DESCRIPTION
**Pull request by saurabh500**
There are two changes in this PR.

1. Handle all the exception thrown by the thread in the thread itself, so that it doesn't crash the tests.
2. Add the check for additional error message that can be thrown if the Command is Canceled. 

This was causing intermittent test crashes in Manual Test runs.

### CODE REVIEW

File: src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs [hunk 1]
```diff
@@ -208,19 +208,28 @@ private static void ExecuteCommandCancelExpected(object state)
...
+                            {
+                            }
+                        } while (r.NextResult());
+                    }
+                }, new string[] { errorMessage, errorMessageSevereFailure });
+            }
+            catch
+            {
+                Assert.True(false, $"An unchecked exception was thrown in {nameof(ExecuteCommandCancelExpected)}");
+            }
```
Comment (stephentoub) [inline src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs]: Why don't we want the original exception to just propagate?  Doing it this way we lose details on what the unexpected exception was.

Comment (saurabh500) [reply src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs]: This function is run on a thread, and when the exception propagates, it crashes the test. I could however print the exception stack trace in the Assert.

Comment (saurabh500) [reply src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs]: The LoC where the thread is being created is https://github.com/saurabh500/corefx/blob/2125de82771c08d64dc63c3b7b57cc551fb8d8de/src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs#L128

Comment (stephentoub) [reply src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs]: Assert.True(false, ...) is also going to throw an exception.

Comment (saurabh500) [reply src/System.Data.SqlClient/tests/ManualTests/SQL/CommandCancelTest/CommandCancelTest.cs]: Ah, yes. I should change this.

**Comment by saurabh500** [comment]
```
System.Net.Http.Functional.Tests.HttpClientHandler_MaxConnectionsPerServer_Test.GetAsync_MaxLimited_ConcurrentCallsStillSucceed(maxConnections: 2, numRequests: 2, secure: False) [FAIL]
12:06:05         System.Threading.Tasks.TaskCanceledException : A task was canceled.
12:06:05         Stack Trace:
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
12:06:05              at System.Net.Http.HttpClient.<FinishSendAsyncBuffered>d__58.MoveNext()
12:06:05           --- End of stack trace from previous location where exception was thrown ---
12:06:05              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
12:06:05              at System.Net.Http.Functional.Tests.HttpClientHandler_MaxConnectionsPerServer_Test.<GetAsync_MaxLimited_ConcurrentCallsStillSucceed>d__3.MoveNext()
12:06:05           --- End of stack trace from previous location where exception was thrown ---
12:06:05              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
12:06:05           --- End of stack trace from previous location where exception was thrown ---
12:06:05              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
12:06:05           --- End of stack trace from previous location where exception was thrown ---
12:06:05              at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
12:06:05              at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
12:06:05     Finished:    System.Net.Http.Functional.Tests

```
NETFX x86 Failure

**Comment by saurabh500** [comment]
Test failure issues 
https://github.com/dotnet/corefx/issues/27834
https://github.com/dotnet/corefx/issues/27835

**Pull request merged**